src/scroll.cc

```diff
@@ -111,11 +108,11 @@ static unsigned int scroll_count_characters_to_right(struct scroll_data* sd, con
 }
 
 static void scroll_scroll_left(struct scroll_data* sd, const std::vector<char>& buf, unsigned int amount) {
-    for (int j = 0; (j < amount) && (buf[sd->start] != '\0'); ++j) {
+    for (int i = 0; (i < amount) && (buf[sd->start] != '\0') && (sd->start < buf.size()); ++i) {
         sd->start += scroll_character_length(buf[sd->start]);
     }
 
-    if(buf[sd->start] == 0 || (unsigned) sd->start > strlen(&(buf[0]))) {
+    if (buf[sd->start] == 0 || sd->start > strlen(buf.data())) {
```

Comment:
In [conky.cc#891](https://github.com/brndnmtthws/conky/blob/master/src/conky.cc#L891)(function `generate_text_internal` that generates content of `buf`) there is a `'\0'` written to the end of buffer, so if there is any unsafe call to `strlen` then that's [conky.cc#885](https://github.com/brndnmtthws/conky/blob/master/src/conky.cc#L885).